app/models/user.rb

```diff
@@ -0,0 +1,49 @@
+# == Schema Information
+#
+# Table name: users
+#
+#  id              :integer          not null, primary key
+#  username        :string           not null
+#  password_digest :string           not null
+#  session_token   :string           not null
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+
+class User < ApplicationRecord
+  validates :username, :password_digest, :session_token, presence: true
+  validates :password, length: { minimum: 6, allow_nil: true }
+
+  attr_reader :password
+
+  after_initialize :ensure_session_token
+
+  has_many :lists
+  has_many :tasks, through: :lists
```

Comment:
This was an leftover that I meant to delete since I wasn't using it. I was thinking of just changing it to a direct `has_many/belongs_to` between users and tasks depending on if it became necessary to achieve a given user story.